src/components/cylc/table/Table.vue

```diff
@@ -239,9 +304,73 @@ export default {
           }
           return true
         })
+      this.sortBy.forEach(sortByProperty => {
+        filteredSortedTasks = filteredSortedTasks.sort((taskA, taskB) => {
+          let valueA
+          let valueB
+          const sortDesc = this.sortDesc[this.sortBy.indexOf(sortByProperty)]
+          switch (sortByProperty) {
+          case 'Task':
+            valueA = typeof taskA.node.name !== 'undefined' ? taskA.node.name : ''
+            valueB = typeof taskB.node.name !== 'undefined' ? taskB.node.name : ''
+            return sortDesc ? valueB.localeCompare(valueA) : valueA.localeCompare(valueB)
+          case 'Cycle Point':
+            valueA = taskA.node.cyclePoint !== '' && typeof taskA.node.cyclePoint !== 'undefined' ? (new Date(taskA.node.cyclePoint)).getTime() : 0
+            valueB = taskB.node.cyclePoint !== '' && typeof taskB.node.cyclePoint !== 'undefined' ? (new Date(taskB.node.cyclePoint)).getTime() : 0
+            return sortDesc ? valueB - valueA : valueA - valueB
+          case 'Jobs':
+            valueA = typeof taskA.jobs !== 'undefined' ? taskA.jobs.length : 0
+            valueB = typeof taskB.jobs !== 'undefined' ? taskB.jobs.length : 0
+            return sortDesc ? valueB - valueA : valueA - valueB
+          case 'Host':
+            valueA = typeof taskA.latestJob.platform !== 'undefined' ? taskA.latestJob.platform : ''
+            valueB = typeof taskB.latestJob.platform !== 'undefined' ? taskB.latestJob.platform : ''
+            return sortDesc ? valueB.localeCompare(valueA) : valueA.localeCompare(valueB)
+          case 'Job System':
+            valueA = typeof taskA.latestJob.jobRunnerName !== 'undefined' ? taskA.latestJob.jobRunnerName : ''
+            valueB = typeof taskB.latestJob.jobRunnerName !== 'undefined' ? taskB.latestJob.jobRunnerName : ''
+            return sortDesc ? valueB.localeCompare(valueA) : valueA.localeCompare(valueB)
+          case 'Job ID':
+            valueA = typeof taskA.latestJob.jobId !== 'undefined' ? taskA.latestJob.jobId : 0
+            valueB = typeof taskB.latestJob.jobId !== 'undefined' ? taskB.latestJob.jobId : 0
+            return sortDesc ? valueB - valueA : valueA - valueB
```

Comment:
As I recall, job IDs are not necessarily numerical for batch systems like PBS and Slurm (gotta check though...)

Reply:
Definately correct about Slurm:

```
sacct -a  -S 09:00 -o "job" | python -c '''
import sys
for line in sys.stdin.readlines():
    try:
        int(line)
    except:
        print(line)
'''
```
gives a load of answers such as
```
123456.bat+   # ch?
123456.0
```

Slurm job array would also probably produce job ids in the form `123456_42`

Reply:
You can ignore the last comment - I don't think that we support slurm job array right now. Trying to use it may well confuse Cylc.